enterprise/views.py

```diff
@@ -1698,22 +1700,47 @@ def post(self, request, enterprise_uuid, course_id):
             # For the audit course modes (audit, honor), where DSC is not
             # required, enroll the learner directly through enrollment API
             # client and redirect the learner to LMS courseware page.
-            if not enterprise_course_enrollment:
-                # Create the Enterprise backend database records for this course enrollment.
-                EnterpriseCourseEnrollment.objects.create(
-                    enterprise_customer_user=enterprise_customer_user,
-                    course_id=course_id,
-                    source=enrollment_source
-                )
-                track_enrollment('course-landing-page-enrollment', request.user.id, course_id, request.get_full_path())
-
+            succeeded = True
             client = EnrollmentApiClient()
-            client.enroll_user_in_course(
-                request.user.username,
-                course_id,
-                selected_course_mode_name,
-                cohort=cohort_name
-            )
+            try:
+                client.enroll_user_in_course(
+                    request.user.username,
+                    course_id,
+                    selected_course_mode_name,
+                    cohort=cohort_name
+                )
+            except HttpClientError as exc:
+                succeeded = False
+                default_message = 'No error message provided'
+                try:
+                    error_message = json.loads(exc.content.decode()).get('message', default_message)
+                except ValueError:
+                    error_message = default_message
+                LOGGER.exception(
+                    'Error while enrolling user %(user)s: %(message)s',
+                    dict(user=self.user_id, message=error_message)
+                )
```

Comment:
Does `str(exc)` not contain a representation of the `message`?

Reply:
You would think it does, right? Here's an example log where this message is extracted out from the exception content in another place within the code:

```
2021-02-19 10:26:56,374 ERROR 27886 [enterprise.models] [user 36854525] [ip 27.122.14.83] models.py:1037 - Error while enrolling user 36854525: An error occurred while creating the new course enrollment for user 'Damiend836' in course 'course-v1:IITBombayX+Linux101.1x+1T2021'
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/enterprise/models.py", line 1027, in enroll
    enrollment_api_client.enroll_user_in_course(self.username, course_run_id, mode, cohort=cohort)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/enterprise/api_client/lms.py", line 104, in inner
    return func(self, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/enterprise/api_client/lms.py", line 229, in enroll_user_in_course
    return self.client.enrollment.post(
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/slumber/__init__.py", line 167, in post
    resp = self._request("POST", data=data, files=files, params=kwargs)
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/slumber/__init__.py", line 101, in _request
    raise exception_class("Client Error %s: %s" % (resp.status_code, url), response=resp, content=resp.content)
slumber.exceptions.HttpClientError: Client Error 400: https://courses.edx.org/api/enrollment/v1/enrollment
```

My understanding is that `LOGGER.exception` still prints out the stack trace even though we don't explicitly tell it to print with `str(exc)`; extracting out the underlying error message from `exc.content` to surface the message seems reasonable here as I think the main exception error/message is coming from slumber (i.e., "Client Error 400") which isn't as useful compared to the underlying error message.

Reply:
Ah, I missed the `LOGGER.exception` - yeah, that'll get you the stack trace.